lib/src/signed_video_h26x_auth.c

```diff
@@ -309,16 +354,21 @@ verify_hashes_with_hash_list(signed_video_t *self, int *num_expected_nalus, int
       uint8_t *expected_hash = &expected_hashes[compare_idx * hash_size];
 
       if (memcmp(hash_to_verify, expected_hash, hash_size) == 0) {
-        // We have a match. Set validation_status and add missing nalus if we have detected any.
-        if (item->second_hash && !item->need_second_verification &&
-            item->nalu->is_first_nalu_in_gop) {
-          // If this |is_first_nalu_in_gop| it should be verified twice. If this the first time we
-          // signal that we |need_second_verification|.
-          DEBUG_LOG("This NALU needs a second verification");
-          item->need_second_verification = true;
+        if (!order_ok) {
+          item->validation_status = 'N';
+          order_ok = true;
```

Comment:
Isn't there a need to update `need_second_verification` and/or `first_verification_not_authentic`?